.github/workflows/build.yml

```diff
@@ -162,8 +162,19 @@ jobs:
         with:
           buildkitd-config-inline: |
             [registry."ghcr.io"]
-      - name: Login to DockerHub
+
+      - name: Check for DockerHub credentials
+        id: docker-credentials
         if: ${{ steps.should.outputs.RUN == 'true' }}
+        run: |
+          if [ -n "${{ secrets.DOCKERHUB_USERNAME }}" ] && [ -n "${{ secrets.DOCKERHUB_TOKEN }}" ]; then
+            echo "do_dockerhub_login=true" >> $GITHUB_OUTPUT
+          else
+            echo "do_dockerhub_login=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Login to DockerHub
+        if: ${{ steps.should.outputs.RUN == 'true' && steps.docker-credentials.outputs.do_dockerhub_login == 'true' }}
```

Comment:
I added this, as it is just for rate limited reasons, but forks don't need to have access to the docker hub creds.